open_schools_platform/organization_management/teachers/migrations/0001_initial.py

```diff
@@ -0,0 +1,134 @@
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        ('circles', '0006_added_history_records'),
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('photos', '0003_added_history_records'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='TeacherProfile',
+            fields=[
+                ('deleted', models.DateTimeField(db_index=True, editable=False, null=True)),
+                ('deleted_by_cascade', models.BooleanField(default=False, editable=False)),
+                ('created_at', models.DateTimeField(db_index=True, default=django.utils.timezone.now)),
+                ('updated_at', models.DateTimeField(auto_now=True)),
+                ('id', models.UUIDField(default=uuid.uuid4, primary_key=True, serialize=False)),
+                ('name', models.CharField(max_length=200)),
+                ('age', models.IntegerField(blank=True, null=True, validators=[django.core.validators.MinValueValidator(0)])),
+                ('phone', phonenumber_field.modelfields.PhoneNumberField(blank=True, max_length=17, null=True, region=None, verbose_name='telephone number')),
+                ('photo', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.SET_NULL, to='photos.photo')),
+                ('user', models.OneToOneField(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='teacher_profile', to=settings.AUTH_USER_MODEL)),
+            ],
+            options={
+                'abstract': False,
+            },
+            bases=(rules.contrib.models.RulesModelMixin, models.Model),
+        ),
+        migrations.CreateModel(
+            name='Teacher',
+            fields=[
+                ('deleted', models.DateTimeField(db_index=True, editable=False, null=True)),
+                ('deleted_by_cascade', models.BooleanField(default=False, editable=False)),
+                ('created_at', models.DateTimeField(db_index=True, default=django.utils.timezone.now)),
+                ('updated_at', models.DateTimeField(auto_now=True)),
+                ('id', models.UUIDField(default=uuid.uuid4, primary_key=True, serialize=False)),
+                ('name', models.CharField(max_length=200)),
+                ('circle', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='teachers', to='circles.circle')),
+                ('teacher_profile', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='teachers', to='teachers.teacherprofile')),
+            ],
+            options={
+                'abstract': False,
+            },
+            bases=(rules.contrib.models.RulesModelMixin, models.Model),
+        ),
+        migrations.CreateModel(
+            name='HistoricalTeacherProfile',
+            fields=[
+                ('deleted', models.DateTimeField(db_index=True, editable=False, null=True)),
```

Comment:
Are HistoricalTeacherProfile and HistoricalTeacher created automatically? Is it normal that they were created? I am just curious because I am a bit out of context)

Reply:
Earlier it was added automatically, but now I corrected it and added it explicitly in the model